### PR TITLE
Allow S3 and GS prefixes to be used as src_uri items

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- Specifying an S3 prefix/folder as a source_uri item (for example:s3://some_bucket/my_data/) is broken, and it looks like it was never implemented for GCS sources. This means we need to pre-generate a tiles.geojson for every data import.
- Pixetl breaks when the ratio of CPUs to tiles is large, such as when running on a subset of a dataset or when resuming a mostly-complete job after our EC2 is recalled

Issue Number: GTC-1331, GTC-1261

## What is the new behavior?

- One can use a gs:// or s3:// URI for src_uris!
- Pixetl will succeed even when the ratio of CPUs to tiles is large

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Also updated quite a lot of dependencies between the data API and pixetl

### CODE REVIEW

File: app/utils/aws.py [hunk 1]
```diff
@@ -67,6 +73,47 @@ async def head_s3(bucket: str, key: str) -> bool:
     return response.status_code == 200
 
 
+def get_aws_files(
```
Comment (jterry64) [inline app/utils/aws.py:76]: If this is blocking, and there's no default limit on the number of files, could someone past a prefix with a bazillion files and just completely block the service?

**Review by jterry64 — APPROVED** [review]
Looks good, just had one question above

File: app/utils/aws.py (hunk 1 above)
Comment (dmannarino) [reply app/utils/aws.py:76]: Mmm, you're right. It's not the RIGHT solution, but for the moment I will add a default limit of 10. This means if you specify a prefix filled with 10 things other than valid files, there's a chance you will get a misleading error message. But hey, they should clean up their junk! Also will create a ticket to lay out everything wrong with this approach.

Comment (dmannarino) [reply app/utils/aws.py:76]: Follow-up ticket: https://gfw.atlassian.net/browse/GTC-1434

**Review by solomon-negusse — APPROVED** [review]

File: app/utils/aws.py (hunk 1 above)
Comment (jterry64) [reply app/utils/aws.py:76]: Oh I missed where you set it! Sounds good.

**Pull request merged**